File: Task3/main.cpp
#include "library.h"
#include <fstream> // Для работы с файлами

int main() {
	std::vector<Tram> trams; // Для варианта 2 (трамваи)

	// Открываем файл для чтения. Предполагаем, что файл называется "input.txt"
	// и находится в той же директории, что и исполняемый файл.
	std::ifstream inputFile("data.conf");

	// Проверяем, удалось ли открыть файл
	if (!inputFile.is_open()) {
		std::cerr << "Не удалось открыть файл input.txt" << std::endl;
		return 1; // Возвращаем код ошибки
	}

	std::string line;
	while (std::getline(inputFile, line)) { // Читаем файл построчно
		std::stringstream ss(line);
		std::string command;
		ss >> command;

		std::string args;
		std::getline(ss, args);				   // Считываем остаток строки как аргументы
		if (!args.empty() && args[0] == ' ') { // Убираем пробел в начале, если есть
			args = args.substr(1);
		}

		processCommand(command, args, trams);
	}

	inputFile.close(); // Закрываем файл после использования

	return 0;
}

File: Task3/Java/Main.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class Main {

	public static void main(String[] args) {
		List<Tram> trams = new ArrayList<>(); // Для варианта 2 (трамваи)

		// Открываем файл для чтения. Убедитесь, что "data.conf" находится в той же директории, что и исполняемый файл.
		try (
			BufferedReader inputFile = new BufferedReader(
				new FileReader("data.conf")
			)
		) {
			String line;
			while ((line = inputFile.readLine()) != null) {
				String[] parts = line.split(" ", 2); // Разделяем команду и аргументы
				String command = parts[0];
				String myArgs = parts.length > 1 ? parts[1] : ""; // Аргументы, если есть

				Library.processCommand(command, myArgs, trams);
			}
		} catch (IOException e) {
			System.err.println(
				"Не удалось открыть файл data.conf: " + e.getMessage()
			);
			System.exit(1); // Завершаем программу с ошибкой
		}
	}
}
